src/lib/components/MarkdownEditor.js

```diff
@@ -1,27 +1,77 @@
 import "./quill.snow.css";
-import React from "react";
-import ReactQuill from "react-quill";
+import React, { useEffect, useRef } from "react";
+
+import Quill from "quill";
 import { sanitizePasteHtml } from "virtualized-chat";
 
 export const MarkdownEditor = (props) => {
-  var toolbarOptions = ['bold', 'italic', 'underline', 'strike', { 'list': 'ordered' }, { 'list': 'bullet' }];
+  const ref = useRef(null);
+  const quillRef = useRef(null);
+  const toolbarOptions = ['bold', 'italic', 'underline', 'strike', { 'list': 'ordered' }, { 'list': 'bullet' }];
 
-  return (
-    <ReactQuill
-      modules={{
-        toolbar: toolbarOptions
-      }}
-      theme="snow"
-      value={props.value}
-      onChange={(value) => {
-        const purifiedText = sanitizePasteHtml(value);
-        props.setValue(purifiedText);
-      }}
-      onKeyDown={(e) => {
-        if (e.key === "Enter" && !e.shiftKey) {
-          props.handleMessageSent();
+  const sendMessage = () => {
+    props.handleMessageSent(quillRef.current.root.innerHTML);
+    quillRef.current.setText('');
+  }
+
+  const handleDrop = (e) => e.preventDefault();
+
+  const handlePaste = (e) => {
+    e.preventDefault();
+    const clipboardData = e.clipboardData;
+    let pastedData;
+
+    if (clipboardData.types.includes('text/html')) {
+      pastedData = clipboardData.getData('text/html');
+    } else if (clipboardData.types.includes('text/plain')) {
+      pastedData = clipboardData.getData('text/plain');
+    }
+    const processedData = sanitizePasteHtml(pastedData);
```

Comment:
do you need to sanitize content if type is text/plain?

Reply:
Yes